### ISSUE TITLE
There should be a tool to help manage and use themes

### ISSUE THREAD

**Issue by lex148** [leftwm/leftwm#27]
some ideas:
```bash
leftwm-themes add https://github.com/lex148/leftwm_theme_bla
leftwm-themes remove leftwm_theme_bla
leftwm-themes update
leftwm-themes update leftwm_theme_bla
leftwm-themes next
leftwm-themes previous
leftwm-themes set leftwm_theme_bla
```

**Comment by stevensonmt** [comment]
Things you've probably already though about:
+ `remove` should have a check that the specified theme is not set as `current` unless a `default` or `fallback` theme is always specified somehow. 
+ Would be nice if `add` and/or `set` commands included a check that necessary utilities were installed. For example if you do `leftwm-themes set basic_polybar` it should check that polybar is installed.

**Comment by Trek3** [comment]
So I tried following those rules @lex148 wrote and created a very basic script that works like this:

- There is a new file in ~/.config/leftwm called "themes.txt" which stores the path of every registered theme
- To register a theme you call ```leftwm-theme add /path/to/theme```
- To remove a theme you call ```leftwm-theme remove /path/to/theme```
- To set a theme you call ```leftwm-theme set /path/to/theme```. The theme must be registered first. Also you need to reload manually leftwm in order to apply the changes.

There is an extra functionality ```leftwm-theme -f``` which scans some basic folders (e.g. ~/.config/leftwm/themes) to find existing themes.
An existing theme is considered valid if there is an up script, a down script and a theme.toml file in the folder.

Of course is a very basic script, but if you are interested I could share.

**Comment by lex148** [comment]
Absolutely I would love to see it.

**Comment by Trek3** [comment]
Ok so please note that I learned bash while writing this so it may not be the best code ever ahah. Of course it needs many improvements and checks and so on. I'd be happy to continue working on this if you like it.

[leftwm-theme.zip](https://github.com/leftwm/leftwm/files/3977872/leftwm-theme.zip)

**Comment by lex148** [comment]
Very nice. So to give you a bit more information about this ticket, I'm envisioning this as almost a package manager for themes. I'm thinking it eventually might have dependency management, voting some sort of "mark as broken" a backend service to search for themes and much more. Because of the scope of this one I have been neglecting starting it. It probably deserves its own Repo. I would love help with it if you are interested in righting some rust code, or if you just want to help brain storm that's cool too.

**Comment by Trek3** [comment]
Well the thing is I have never written code in Rust, and looking through your code for various reasons I can say I barely understand what is happening in the code, so I'm afraid I can't be very helpful. At least not now, because I don't have time to learn Rust enough. 

The idea is actually very cool and should you start it some day I'll try to help. 
Let me ask you one thing for my script anyway: how would you call SoftReload command from command line?

**Comment by TerminalWitchcraft** [comment]
Hey @lex148 , the idea of a theme manager seems interesting! I would like to create a rust based application for it! Do you have any specific opinions/ideas about it except the ones you already mentioned? I can probably pull up a basic theme manager within few weeks, which does all setup/teardown phases in a safe and reliable way(i.e, it will check if program is installed/running, etc). Maybe I can start working on a repo which can be later merged to this organization if its stable and worthy. Let me know your thoughts on this. 

Best,
Hitesh

P.S- Thank you for the project! I envision this as a stable rock solid alternative to other wms, especially xmonad, where it shines as the one which you can trust and make sure it doesn't break! I'm still need to explore leftwm in more depth, but your initial design goals looks promising! It would also be great if we have a community communication platform, like irc/discord/gittter whatever. That would make communication much easier!

**Comment by lex148** [comment]
@TerminalWitchcraft that sounds awesome! I am very open to ideas on this one. The only one hard requirement that I have is I think LeftWM should still work with a theme even if the theme-manager is not installed. I would like to help out two if you are open to working as a team.

**Comment by TerminalWitchcraft** [comment]
@lex148 yes, I agree with you, leftwm should be independent of the theme manager. And yes, I'm would love to collaborate! What would you prefer for the repository? Do you want to create a repository under leftwm and add me as a collaborator or I can create one and add you(and anyone else who would like to work) to the repository. Besides, do you have any name in mind for the binary? I was thinking of leftwm-theme but I'm open to options.

As a side note, I was thinking we can create the binary which could read themes from a Git repo. For instance `leftwm-theme apply github.com/user/theme`. It's better if all theme data followed XDG base directories structure. We can create a github template to help more users start off creating their own rice! Themes can be updated by `leftwm-theme update themename`. This is just an initial idea, but let me know your thoughts on this.

**Comment by lex148** [comment]
I have sent you an invite to the repo. Let me know if you have any problems. 

I really like the idea of reading a theme from a git repo. Cant wait to see what you come up with.

**Comment by TerminalWitchcraft** [comment]
The repo looks good! I have described an initial outline for the project in the readme file under `develop` branch. Any suggestions or ideas would be welcomed! Let's carry on further discussions related to the theme-manager over there.

**Comment by mautamu** [comment]
Closed in leftwm/leftwm-theme#4
